runelite-client/src/main/java/net/runelite/client/plugins/worldhopper/WorldHopperPlugin.java

```diff
@@ -0,0 +1,538 @@
+import net.runelite.client.util.HotkeyListener;
+import net.runelite.client.util.SwingUtil;
+import net.runelite.client.util.Text;
+import net.runelite.http.api.worlds.World;
+import net.runelite.http.api.worlds.WorldClient;
+import net.runelite.http.api.worlds.WorldResult;
+import net.runelite.http.api.worlds.WorldType;
+import org.apache.commons.lang3.ArrayUtils;
+
+@PluginDescriptor(
+	name = "World Hopper",
+	description = "Allows you to quickly hop worlds"
+)
+@Slf4j
+public class WorldHopperPlugin extends Plugin
+{
+	private static final String HOP_TO = "Hop-to";
+	private static final String KICK_OPTION = "Kick";
+	private static final ImmutableList<String> BEFORE_OPTIONS = ImmutableList.of("Add friend", "Remove friend", KICK_OPTION);
+	private static final ImmutableList<String> AFTER_OPTIONS = ImmutableList.of("Message");
+
+	@Inject
+	private Client client;
+
+	@Inject
+	private ClientThread clientThread;
+
+	@Inject
+	private ConfigManager configManager;
+
+	@Inject
+	private ClientToolbar clientToolbar;
+
+	@Inject
+	private KeyManager keyManager;
+
+	@Inject
+	private ChatMessageManager chatMessageManager;
+
+	@Inject
+	private ScheduledExecutorService executorService;
+
+	@Inject
+	private WorldHopperConfig config;
+
+	private NavigationButton navButton;
+	private WorldSwitcherPanel panel;
+
+	private int favoriteWorld1, favoriteWorld2;
+
+	private final HotkeyListener previousKeyListener = new HotkeyListener(() -> config.previousKey())
+	{
+		@Override
+		public void hotkeyPressed()
+		{
+			hop(true);
+		}
+	};
+
+	private final HotkeyListener nextKeyListener = new HotkeyListener(() -> config.nextKey())
+	{
+		@Override
+		public void hotkeyPressed()
+		{
+			hop(false);
+		}
+	};
+
+	private Future<?> worldResultFuture;
+	private WorldResult worldResult;
+
+	@Provides
+	WorldHopperConfig getConfig(ConfigManager configManager)
+	{
+		return configManager.getConfig(WorldHopperConfig.class);
+	}
+
+	@Override
+	protected void startUp() throws Exception
+	{
+		keyManager.registerKeyListener(previousKeyListener);
+		keyManager.registerKeyListener(nextKeyListener);
+
+		worldResultFuture = executorService.submit(() ->
+		{
+			try
+			{
+				WorldResult worldResult = new WorldClient().lookupWorlds();
+
+				if (worldResult != null)
+				{
+					worldResult.getWorlds().sort(Comparator.comparingInt(World::getId));
+					this.worldResult = worldResult;
+
+					SwingUtilities.invokeLater(() -> panel.populate(worldResult.getWorlds()));
+				}
+			}
+			catch (IOException ex)
+			{
+				log.warn("Error looking up worlds", ex);
+			}
+		});
+
+		panel = new WorldSwitcherPanel(this);
+
+		final BufferedImage icon;
+		synchronized (ImageIO.class)
+		{
+			icon = ImageIO.read(getClass().getResourceAsStream("icon.png"));
+		}
+
+		navButton = NavigationButton.builder()
+			.tooltip("World Switcher")
+			.icon(icon)
+			.priority(3)
+			.panel(panel)
+			.build();
+
+		clientToolbar.addNavigation(navButton);
```

Comment:
This entire panel should be disable-able, as I would at least see features that reimplement parts of game be optional. The shortcuts are still nice to use even without the panel.